src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl_ModelComputation.cs

```diff
@@ -210,5 +232,12 @@ private async ValueTask HighlightExpandAndPresentItemsAsync(ImmutableSegmentedLi
 
             return _editorAdaptersFactoryService.GetWpfTextView(textView);
         }
+
+        private void UpdateProgressBarVisibility(Visibility visibility)
+        {
+            _threadingContext.ThrowIfNotOnUIThread();
+            if (ProgressBar.Visibility != visibility)
```

Comment:
Is this check necessary? I'd assume just assigning the same value will do any optimizations under the covers.

Reply:
This code is now outdated but the idea was that the UI should only be notified of a property change when the value is actually changed. This should be reflected in the updated code and I've added a comment.